src/code_806FDF4.c

```diff
@@ -0,0 +1,1092 @@
+            {
+                info->shopkeeper = 2;
+            }
+            else
+            {
+                info->shopkeeper = 3;
+            }
+        }
+    }
+}
+
+void sub_806F480(Entity *pokemon, u8 r1)
+{
+    EntityInfo *info;
+
+    info = GetEntInfo(pokemon);
+
+    if(info->shopkeeper != 0)
+    {
+        if(r1)
+        {
+            info->shopkeeper = 2;
+        }
+        else
+        {
+            info->shopkeeper = 3;
+        }
+    }
+}
+
+u8 sub_806F4A4(Entity *pokemon, u8 type) {
+
+    EntityInfo *info = pokemon->info;
+    s32 index;
+    
+    if(MonsterIsType(pokemon, TYPE_GHOST))
+    {
+        if(type == TYPE_NORMAL || type == TYPE_FIGHTING)
+            if(!info->exposed)
+                return FALSE;
+    }
+    for(index = 0; index < 2; index++)
+    {
+        if(gTypeEffectivenessChart[type][info->types[index]] == EFFECTIVENESS_IMMUNE)
+            return FALSE;
+    }
+    return TRUE;
+}
+
+void sub_806F500(void)
+{
+    struct unkStruct_Dungeon134_sub *temp;
+
+    temp = &gDungeon->unk134;
+
+    temp->unk134 = 0;
+    temp->unk138 = 0;
+    temp->unk13C[0] = 0;
+    temp->unk13C[1] = 0;
+    temp->unk13E[0] = 0xA;
+    temp->unk13E[1] = 0xA;
+    temp->unk140[0] = 1;
+    temp->unk140[1] = 1;
+    temp->unk140[2] = 0;
+    temp->unk140[4] = 0;
+    temp->unk140[3] = 0;
+    temp->unk140[5] = 0;
+    temp->unk140[6] = 0;
+    temp->unk150 = 0;
+    temp->unk158 = 0;
+    temp->unk15C = 0;
+    temp->unk160 = 0;
+    temp->unk161 = 0;
+    temp->unk162 = 0;
+    temp->unk163 = 0;
+    temp->unk164 = 0;
+    temp->unk165 = 0;
+    temp->unk166 = 0;
+    temp->unk167 = 0;
+    temp->unk168 = 0;
+    temp->unk169 = 0;
+    temp->unk16A = 0;
+    temp->unk16C = 0;
+    temp->unk16B = 0;
+    temp->unk16D = 0;
+    temp->fill16E[0] = 0;
+    temp->fill16E[1] = 0;
+    temp->fill16E[2] = 0;
+    temp->fill16E[3] = 0;
+    temp->fill16E[4] = 0;
+    temp->fill16E[5] = 0;
+    temp->fill16E[6] = 0;
+    temp->fill16E[7] = 0;
+    temp->fill16E[8] = 0;
+    temp->fill16E[9] = 0;
+    temp->fill16E[10] = 0;
+    temp->pokemonExposed = FALSE;
+    temp->unk17A = 0;
+}
+
+s32 sub_806F62C(int param_1)
+{
+    return gUnknown_8106F7C[param_1];
+}
+
+void sub_806F63C(Entity *param_1)
+{
+    UnkDungeonGlobal_unk181E8_sub *temp;
+
+    temp = &gDungeon->unk181e8;    
+
+    if (temp->cameraTarget == param_1) {
+        sub_803F508(temp->cameraTarget);
+    }
+}
+
+// https://decomp.me/scratch/2la6b - (96.72% matching - Seth)
+#ifdef NONMATCHING
+bool8 sub_806F660(Entity *pokemon,Entity *target)
+{
+  EntityInfo *pokemonInfo;
+  EntityInfo *targetInfo;
+  s32 size;
+  int iVar4;
+  s32 iVar5;
+  s32 uVar6;
+
+  int recruitRate;
+  s32 iVar8;
+  s32 index;
+  pokemonInfo = pokemon->info;
+  targetInfo = target->info;
+  iVar8 = -1;
+  size = GetBodySize(targetInfo->apparentID);
+  if ((1 < ((u16)(gDungeon->bossBattleIndex - 4))) && (gDungeon->bossBattleIndex != 9) && (gDungeon->bossBattleIndex != 0xf)) {
```

Comment:
This comparison has broken me. I no longer know what kind of horrendous comparison macros they could have used.